pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+
+  </div>
+</details>
+
+
+## Bootstrapping core
+
+In order to have anything working at all in our pages, we need to at least bootstrap *@pyscript/core* functionalities, otherwise all examples and scripts mentioned in this document would just sit there ... sadly ignored by every browser:
+
+```html
+<!doctype html>
+<html>
+    <head>
+        <!-- this is a way to automatically bootstrap @pyscript/core -->
+        <script type="module" src="https://esm.run/@pyscript/core"></script>
+    </head>
+    <body>
+        <script type="micropython">
+            from js import document
+            document.body.textContent = '@pyscript/core'
+        </script>
+    </body>
+</html>
+```
+
+As *core* exposes some utility/API, using the following method would also work:
+
+```html
+<script type="module">
+    import {
+        define,      // define a custom type="..."
+        whenDefined, // wait for a custom type to be defined
+        XWorker      // allows JS <-> Interpreter communication
+    } from 'https://esm.run/@pyscript/core';
+</script>
+```
+
+Please keep reading this document to understand how to use those utilities or how to have other *Pogramming Languages* enabled in your page via `<script>` elements.
+
+
+## How Scripts Work
+
+The [&lt;script&gt; element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) has at least these extremely important peculiarities compared to any other element defined by the [HTML Standard](https://html.spec.whatwg.org/multipage/):
+
+ * its only purpose is to contain *data blocks*, meaning that browsers will never try to parse its content as generic *HTML* (and browsers will completely ignore either its content or its attributes, including the `src`, when its [type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type) is not known)
+ * its completely unobtrusive when it comes to both *aria* and *layout*, indeed it's one of the few nodes that can be declared almost anywhere without breaking its parent tree (other notable exception would be a comment node)
+ * for our specific use case, it already offers attributes that are historically well understood and known, also simplifying somehow the creation of this document
+
+The long story short is that any `<script type="non-standard-type">` has zero issues with any browser of choice, but it's true that using some specific *custom type* might lead to future issues in case that `type` could have some special meaning for the future of the Web.
+
+We encourage everyone to be careful when using this *core* API as we definitively don't want to clash or conflict, by any mean, with what the Web might need or offer in the near to far future, but we're also confident so far our current *types* are more than safe.
```

Comment:
```suggestion
While`<script type="non-standard-type">` works with any browser, it's possible that some of the currently non-standard-types become standard in the future. For this reason, we've chosen to use specific names which we feel are safe (i.e. `<script type="pyodide">` instead of `type="python"`) - users are encouraged to use the same caution when specifying their own type names to avoid conflicts with future standards.
```

Reply:
Probably worth mentioning that we "explicitly take that into account when adding default types in core and any plugin adding aliases or support for new `types` should that that into consideration as well", or something like that